### PR TITLE
Remove CheckAndSetDefaults from types.Resource

### PR DESCRIPTION
**Pull request by rosstimothy**
Related https://github.com/gravitational/teleport.e/pull/2874

### CODE REVIEW

**Review by hugoShaka — APPROVED** [review]

**Review by strideynet — APPROVED** [review]

**Review by codingllama — APPROVED** [review]
Many thanks for tackling this!

File: lib/services/kubernetes.go [hunk 1]
```diff
@@ -65,17 +65,17 @@ type Kubernetes interface {
...
 		return nil, trace.Wrap(err)
 	}
 
 	switch server := kubeServer.(type) {
 	case *types.KubernetesServerV3:
+		if err := server.CheckAndSetDefaults(); err != nil {
+			return nil, trace.Wrap(err)
+		}
+
 		if !cfg.PreserveResourceID {
```
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/services/kubernetes.go:79]: (Doesn't have to happen in this PR.)

**Comment by codingllama** [comment]
Do you think it's worth backporting, so there's less line drift in those files?

File: lib/services/kubernetes.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/services/kubernetes.go:79]: Many thanks!

**Pull request merged**